### PR TITLE
Fix: update eslint-config-amex to v12

### PR DESCRIPTION
**Pull request by PixnBits**
## Description

Update eslint-config-amex to ^12
Automatic and manual fixes applied.

## Motivation and Context

Prevent more issues!

## How Has This Been Tested?

locally `$ npm run lint -- --fix` and `$ npm test`

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [x] My changes are in sync with the code style of this project.
- [x] There aren't any other open Pull Requests for the same issue/update.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] This change impacts caching for client browsers.
- [ ] This change adds additional environment variable requirements for Holocron users.
- [ ] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using Holocron?

Gain the assurances of the rules introduced in eslint-config-amex@12

### CODE REVIEW

File: scripts/check-versions.js [hunk 1]
```diff
@@ -34,16 +34,19 @@ function parseVersion(raw) {
     return raw;
   }
 
-  const match = /^v?(\d+)\.(\d+)\.(\d+)(?:-(.+))?$/.exec(raw.trim());
```
Comment (PixnBits) [inline scripts/check-versions.js:37]: split out the parsing due to [unicorn/no-unsafe-regex](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-unsafe-regex.md) calling out the `(?:-(.+))?` bit

File: packages/holocron/src/holocronModule.jsx [hunk 2]
```diff
@@ -191,8 +195,8 @@ export default function holocronModule({
     hoistStatics(HolocronModuleWrapper, WrappedComponent);
 
     const DerivedComponent = connect(
-      mapModuleStateToProps,
-      mapDispatchToProps,
+      mapModuleStateToProperties,
+      mapDispatchToProperties,
```
Comment (JAdshead) [inline packages/holocron/src/holocronModule.jsx:199]: this goes against convention..

File: package.json [hunk 3]
```diff
@@ -39,7 +39,7 @@
     "enzyme": "^3.11.0",
     "enzyme-to-json": "^3.2.2",
     "eslint": "^6.8.0",
-    "eslint-config-amex": "^11.2.0",
+    "eslint-config-amex": "^12.2.0",
```
Comment (JAdshead) [inline package.json:42]: I am not a fan of `props` to `properties`, I believe that this lint rule is removed in later versions.

Comment (PixnBits) [reply package.json:42]: Ah! Though that's what "props" is short for I can add an ignore directive before the update to v13 👍

File: packages/holocron/src/holocronModule.jsx (hunk 2 above)
Comment (PixnBits) [reply packages/holocron/src/holocronModule.jsx:199]: good catch!!

**Comment by PixnBits** [comment]
need to restart due to the `prop` bit (lots of changes)
maybe it'd be better to move to v13? 🤔

**Comment by PixnBits** [comment]
#157 opened instead

**Pull request closed without merging**